app/src/main/java/de/danoeh/antennapod/fragment/SubscriptionFragment.java

```diff
@@ -0,0 +1,108 @@
+package de.danoeh.antennapod.fragment;
+
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.GridView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import de.danoeh.antennapod.R;
+import de.danoeh.antennapod.activity.MainActivity;
+import de.danoeh.antennapod.adapter.NavListAdapter;
+import de.danoeh.antennapod.adapter.SubscriptionsAdapter;
+import de.danoeh.antennapod.core.feed.Feed;
+import de.danoeh.antennapod.core.storage.DBReader;
+import de.greenrobot.event.EventBus;
+import rx.Observable;
+import rx.functions.Action1;
+
+/**
+ * Fragment for displaying feed subscriptions
+ */
+public class SubscriptionFragment extends Fragment {
+
+    public static final String TAG = "SubscriptionFragment";
+
+    private GridView mSubscriptionGridLayout;
+    private DBReader.NavDrawerData mDrawerData;
+    private SubscriptionsAdapter mSubscriptionAdapter;
+    private NavListAdapter.ItemAccess mItemAccess;
+
+    private List<Feed> mSubscriptionList = new ArrayList<>();
+
+
+    public SubscriptionFragment() {
+    }
+
+
+    public void setItemAccess(NavListAdapter.ItemAccess itemAccess) {
+        mItemAccess = itemAccess;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setRetainInstance(true);
```

Comment:
You should register with the EventDistributor and react to EventDistributor.FEED_LIST_UPDATE (feeds might be downloaded in a background thread and you probably want to display them as soon as they are processed)